### PR TITLE
localsend: 1.15.4-unstable-2024-10-26 -> 1.16.1

### LINKED ISSUE #354044: localsend: goes black after sending a file
### Describe the bug
LocalSend's window fades to black after a file is sent.

### Steps To Reproduce
1. Go into the Send menu.
2. Pick a file & send it to another device.
3. Click Done.

### Expected behavior
I should be sent back to the Send menu.

### Screenshots
A screenshot of the LocalSend window, all black

### Additional context
Not reproducible with Flatpak (tested).

### Notify maintainers

@sikmir
@linsui
@pandapip1

### Metadata

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ghost**
- Fixes #354044
- https://github.com/localsend/localsend/releases/tag/v1.16.0
- https://github.com/localsend/localsend/releases/tag/v1.16.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/lo/localsend/package.nix [hunk 1]
```diff
@@ -11,27 +11,57 @@
...
+            runHook preInstall
+            chmod +x cargokit/run_build_tool.sh
+            mkdir $out
+            cp -a ./* $out/
+            runHook postInstall
+          '';
+        };
+    };
+
+    cargoDeps = rustPlatform.importCargoLock {
```
Comment (linsui) [inline pkgs/by-name/lo/localsend/package.nix:61]: Please use fetchCargoTarball here.

File: pkgs/by-name/lo/localsend/package.nix [hunk 2]
```diff
@@ -40,10 +70,25 @@ let
...
+      jq
+      cargo
+      rustc
+      rustPlatform.cargoSetupHook
     ];
 
     buildInputs = [ libayatana-appindicator ];
 
+    postPatch = ''
+      cp ${./Cargo.lock} ./Cargo.lock
```
Comment (linsui) [inline pkgs/by-name/lo/localsend/package.nix:83]: Is this file copied from https://github.com/Tienisto/rhttp/blob/main/rhttp/rust/Cargo.lock?

File: pkgs/by-name/lo/localsend/package.nix [hunk 3]
```diff
@@ -11,27 +11,68 @@
...
-      hash = "sha256-qaW9LoRQge4K/CAQwbjNxmlUkQmxvaZyBJdAjXbbTw4=";
+      rev = "v${version}";
+      hash = "sha256-0FBHT7hIDVpRVc+a1Y7g35MGzoE66oGf0CvsU1dUqCc=";
     };
 
     sourceRoot = "${src.name}/app";
 
     pubspecLock = lib.importJSON ./pubspec.lock.json;
 
+    customSourceBuilders = {
```
Comment (linsui) [inline pkgs/by-name/lo/localsend/package.nix:40]: Please put this in https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/compilers/dart/package-source-builders, same for hiddify...

**Comment by sikmir** [comment]
1.16.1 is out

**Comment by sikmir** [comment]
> Should we continue making changes here

Yes, that's fine.

**Review by Pandapip1 — APPROVED** [review]
Diff lgtm; app launches and seems to work correctly. I'm having issues with getting my devices to talk to one another on my university's WiFi though, so I can't test file transfers ATM.

**Review by Pandapip1 — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**